lib/ansible/modules/dnf.py

```diff
@@ -537,6 +537,8 @@ def _configure_base(self, base, conf_file, disable_gpg_check, installroot='/', s
         conf.sslverify = sslverify
 
         # Set installroot
+        if os.path.isfile(installroot):
```

Comment:
```suggestion
        if not os.path.isdir(installroot):
```

We need to check specifically for a directory. We can't assume that if the path is not a regular file that it is a directory. The path can be non-existing, broken symlink, ... Same below.